.github/workflows/codeql-analysis.yml

```diff
@@ -92,76 +82,28 @@ jobs:
       - name: Check out current commit (${{ github.sha }})
         uses: actions/checkout@v2
       - name: Set up Node
-        uses: actions/setup-node@v1
-        # we use a hash of yarn.lock as our cache key, because if it hasn't changed, our dependencies haven't changed,
-        # so no need to reinstall them
-      - name: Compute dependency cache key
-        id: compute_lockfile_hash
-        run: echo "hash=${{ hashFiles('yarn.lock') }}" >> "$GITHUB_OUTPUT"
-      - name: Check dependency cache
-        uses: actions/cache@v2
-        id: cache_dependencies
+        uses: actions/setup-node@v4
         with:
-          path: ${{ env.CACHED_DEPENDENCY_PATHS }}
-          key: ${{ steps.compute_lockfile_hash.outputs.hash }}
+          node-version: 22
+          cache: 'yarn'
       - name: Install dependencies
-        if: steps.cache_dependencies.outputs.cache-hit == ''
         run: yarn install
-    outputs:
-      dependency_cache_key: ${{ steps.compute_lockfile_hash.outputs.hash }}
-
-  job_build:
```

Comment:
why we don't need this anymore?

Reply:
all this step does is run `npm build` which isn't worth caching since it only takes a second and the dependency cache is now managed by `actions/setup-node@v4`

Reply:
this action could be simplified even further. before it had 3 steps install + cache, build + cache, run codeql. Really we could just do install build codeql as a single step.